### PR TITLE
ROX-23299: Platform CVEs view to query db for PlatformCVEs pages

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

Added a view interface to query db for PlatformCVEs. This PR adds queries to support PlatformCVEs list and single pages. GraphQL resolvers and support for clusters list and single pages will be added in followup PRs.

This PR also updates the cluster proto and adds a search tag to cluster type. The migration for the same will be added in a followup PR.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Unit tests

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/views/common/sac.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// in the given context's access scope
+func WithSACFilter(ctx context.Context, targetResource permissions.ResourceMetadata, query *v1.Query) (*v1.Query, error) {
+	var sacQueryFilter *v1.Query
+
+	scopeChecker := sac.GlobalAccessScopeChecker(ctx).AccessMode(storage.Access_READ_ACCESS).Resource(targetResource)
+	scopeTree, err := scopeChecker.EffectiveAccessScope(permissions.View(targetResource))
+	if err != nil {
+		return nil, err
+	}
+	sacQueryFilter, err = sac.BuildNonVerboseClusterNamespaceLevelSACQueryFilter(scopeTree)
```
Comment (md2119) [inline central/views/common/sac.go:23]: Platform vulnerabilities are evaluated against the cluster only. Use `sac.BuildNonVerboseClusterLevelSACQueryFilter`

File: central/views/platformcve/db_response.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	return c.Openshift4ClusterCount
+}
+
+type clusterResponse struct {
+	ClusterID string `db:"cluster_id"`
+
+	// Following are supported sort options.
+	ClusterName string              `db:"cluster"`
+	ClusterType storage.ClusterType `db:"platform_type"`
+	// TODO(charmik): Add support for sorting by k8s version
```
Comment (md2119) [inline central/views/platformcve/db_response.go:101]: We only show high-level types. I would argue with UX the value addition of sorting. Filtering would address the use case.

File: proto/storage/cluster.proto [hunk 3]
```diff
@@ -143,7 +143,7 @@ message SensorDeploymentIdentification {
 message Cluster {
   string id = 1; // @gotags: search:"Cluster ID,hidden,store" sql:"pk,type(uuid)"
   string name = 2; // @gotags: search:"Cluster,store" sql:"unique"
-  ClusterType type = 3;
```
Comment (md2119) [inline proto/storage/cluster.proto:146]: Could you pull out the schema change into a separate PR and push it with the respective migration?

**Review by md2119 — DISMISSED** [review]
The SAC is done incorrectly, otherwise LGTM.

File: central/views/platformcve/view_impl.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+					AddExactMatches(
+						search.ClusterPlatformType,
+						storage.ClusterType_OPENSHIFT4_CLUSTER.String(),
+					).ProtoQuery(),
+			).Proto(),
+	)
+	return cloned
+}
+
+func withCountQuery(q *v1.Query) *v1.Query {
```
Comment (c-du) [inline central/views/platformcve/view_impl.go:169]: https://github.com/stackrox/stackrox/blob/b75e0acab8600b0b76247744c2d27acd126b512a/central/views/imagecve/view_impl.go#L330
Would you like to dedupe and move it to common?

File: proto/storage/cluster.proto (hunk 3 above)
Comment (charmik-redhat) [reply proto/storage/cluster.proto]: Done

File: central/views/platformcve/db_response.go (hunk 2 above)
Comment (charmik-redhat) [reply central/views/platformcve/db_response.go]: I think we can keep the sorting because it shouldn't cause any performance issues.

**Comment by charmik-redhat** [comment]
> Overall LGTM. Just one question, I saw the jira include the field "ClusterVulnerability" which is not included in this PR. What is it? Is it the severity?

I added ClusterVulnerability to mean the ClusterCVE proto object. I think I will add code to fetch the serialized column and unmarshal in a graphQL query and not in this view. That way we can only fetch serialized if requested by user.

File: central/views/common/sac.go (hunk 1 above)
Comment (charmik-redhat) [reply central/views/common/sac.go:23]: Done

File: central/views/platformcve/view_impl.go (hunk 4 above)
Comment (charmik-redhat) [reply central/views/platformcve/view_impl.go:169]: Good idea. Moved it to common.
